src/View/Sidebar/DeviceListBox.vala

```diff
@@ -58,38 +48,71 @@ public class Sidebar.DeviceListBox : Gtk.ListBox, Sidebar.SidebarListInterface {
                 select_item ((SidebarItemInterface) row);
             }
         });
+
+        set_sort_func (device_sort_func);
     }
 
-    private DeviceRow? add_bookmark (string label, string uri, Icon gicon,
+    private int device_sort_func (Gtk.ListBoxRow? row1, Gtk.ListBoxRow? row2) {
+        string key1 = row1 != null ? ((AbstractMountableRow)row1).sort_key : "";
+        string key2 = row2 != null ? ((AbstractMountableRow)row2).sort_key : "";
```

Comment:
Shouldn't we also check for the type here? Also, we can simply use `var`:

```
var key 1 = row1 != null && (row1 is AbstractMountableRow) ? ...
...
```

Reply:
Done